iroh/src/endpoint/quic.rs

```diff
@@ -484,3 +543,177 @@ impl QuicTransportConfig {
         self
     }
 }
+
+/// Parameters governing incoming connections
+///
+/// Default values should be suitable for most internet applications.
+// Note: used in `iroh::endpoint::connection::Incoming::accept_with`
+// This is new-typed since `quinn::ServerConfig` takes a `TransportConfig`, which we new-type as a `QuicTransportConfig`
+#[derive(Debug, Clone)]
+pub struct ServerConfig {
+    inner: quinn::ServerConfig,
+    transport: QuicTransportConfig,
+}
+
+impl ServerConfig {
+    /// Transport configuration to use for incoming connections
+    pub fn transport(&self) -> QuicTransportConfig {
+        self.transport.clone()
+    }
+
+    /// TLS configuration used for incoming connections
+    ///
+    /// Must be set to use TLS 1.3 only.
+    pub fn crypto(&self) -> Arc<dyn CryptoServerConfig> {
+        self.inner.crypto.clone()
+    }
+
+    /// Configuration for sending and handling validation tokens
+    pub fn validation_token(&self) -> ValidationTokenConfig {
+        self.inner.validation_token.clone()
+    }
+
+    pub(crate) fn to_arc(&self) -> Arc<quinn::ServerConfig> {
+        Arc::new(self.inner.clone())
+    }
+
+    /// Create a default config with a particular handshake token key
+    pub fn new(crypto: Arc<dyn CryptoServerConfig>, token_key: Arc<dyn HandshakeTokenKey>) -> Self {
+        let mut inner = quinn::ServerConfig::new(crypto, token_key);
+        let transport = QuicTransportConfig::default();
+        inner.transport_config(transport.to_arc());
+        Self { inner, transport }
+    }
+
+    /// Set a custom [`QuicTransportConfig`]
```

Comment:
```suggestion
    /// Sets a custom [`QuicTransportConfig`].
```
because I can't help the nitpicking